### PR TITLE
92 new tests

### PR DESCRIPTION
**Pull request by dissolve**
this updates all the tests to match the microformats/tests test set.

Previously i had updated the tests to reflect what I believed to be the correct result or to make sure the tests were consistent with current output.  Instead i am just marking any failed ones as pending.

Additionally, and more importantly, this updates for the implied name and implied url parser changes.

I would recommend we move to 4.0.8 after this merge

### CODE REVIEW

File: lib/microformats/format_parser.rb [hunk 1]
```diff
@@ -7,6 +7,7 @@ def parse(element, base: nil, element_type: nil, format_class_array: [], backcom
 
       @properties = {}
       @children = []
+      @seen_types = {}
```
Comment (jgarber623) [inline lib/microformats/format_parser.rb]: What does `@seen_types` refer to?

Comment (dissolve) [reply lib/microformats/format_parser.rb]: this is an internal variable to note all the prefixes seen in a microformat.  The change to when to add implied url or name requires somewhere to track this

Comment (dissolve) [reply lib/microformats/format_parser.rb]: also, yes, totally cool with you documenting and updating

Comment (dissolve) [reply lib/microformats/format_parser.rb]: just pushed updated test set as well

Comment (jgarber623) [reply lib/microformats/format_parser.rb]: > this is an internal variable to note all the prefixes seen in a microformat. The change to when to add implied url or name requires somewhere to track this

Ah, okay, so it's part of the implied properties bits. I need to do a bit more reading on the wiki to fully understand that.

Would a better name for this be something like `matched_prefixes` or `found_prefixes` or something similar…? Or is "type" the appropriate nomenclature for `h-`, `p-`, `u-`, etc. ?

Comment (dissolve) [reply lib/microformats/format_parser.rb]: found_prefixes makes more sense certainly.  

Using set / subset works for me.
I am not good at all the fancier ways of doing things in ruby, so i'll take any suggestions certainly.  That looks like a much cleaner way to go

Comment (dissolve) [reply lib/microformats/format_parser.rb]: > Ah, okay, so it's part of the implied properties bits. I need to do a bit more reading on the wiki to fully understand that.

It changed 

> if no explicit "name" property 

to 

> if no explicit "name" property, and no other p-* or e-* properties, and no nested microformats, 

and similarly  for url 

> if no explicit "url" property,

to 

> if no explicit "url" property, and no other explicit u-* properties, and no nested microformats,

Comment (jgarber623) [reply lib/microformats/format_parser.rb]: > I am not good at all the fancier ways of doing things in ruby, so i'll take any suggestions certainly. That looks like a much cleaner way to go

Not to worry! `Set` is pretty new to me, too. I happened upon it while looking at ways we might clean up `@seen_types`. I can take a swing at the conversion and if it works out will push up a commit.

**Comment by jgarber623** [comment]
Oh, and if the intent is to release 4.0.8, make sure to bump `version.rb`, create the git tag, etc. etc. etc.

File: lib/microformats/format_parser.rb [hunk 2]
```diff
@@ -278,6 +283,15 @@ def parse_element(element)
       bc_classes_found = false
       fmt_classes = format_classes(element)
 
+      prop_classes.each do |p_class|
+        element_type = p_class.downcase.split('-')[0]
```
[the remark this answers is not in the document]
Comment (dissolve) [reply lib/microformats/format_parser.rb]: my concern is mucking about too much all at once, i'm fine doing this type of clean-up sweep of the code before 4.0.8, but i would rather not have this PR get too long in the tooth

**Pull request merged**